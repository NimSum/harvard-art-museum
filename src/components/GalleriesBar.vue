<template>
  <aside>
    <nav>
      <ul>
        <li v-for="gallery in galleries" v-bind:key="gallery.galleryid">
          {{ gallery.theme || gallery.name }}
        </li>
      </ul>
    </nav>
  </aside>
</template>

<script>
export default {
  name: 'GalleriesBar',
  props: ['galleries']
}
</script>

<style lang="scss" scoped>
    aside {
      width: 25%;

      ul {
        margin: 0;
        padding: 10px;

        li {
          list-style-type: none;
          text-align: left;
          padding: 10px;
          margin: 0;
          font-size: 1.2rem;
        }
      }
    }
</style>
