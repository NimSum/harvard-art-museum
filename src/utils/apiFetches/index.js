import axios from 'axios';

const getAnything = async (url) => {
  console.log('fetching')
  const response = await axios.get(url);
  if (response.statusText !== 'OK') {
    throw new Error('Failed to fetch data');
  }
  return response.data;
}

export default getAnything;