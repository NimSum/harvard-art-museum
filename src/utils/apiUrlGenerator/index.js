const apiUrls = {
  galleriesByFloor: (floor) => `https://api.harvardartmuseums.org/gallery?floor=${floor}&hasimage=1&apikey=${process.env.VUE_APP_API_KEY}&size=15&fields=galleryid,name,theme,objectcount,labeltext`,
  getGalleryObjects: (galleryId) => `https://api.harvardartmuseums.org/object?apikey=${process.env.VUE_APP_API_KEY}&sort=random&gallery=${galleryId}`,
  randomImages: (randomStart) => `https://api.harvardartmuseums.org/image?apikey=${process.env.VUE_APP_API_KEY}&size=21&sort=random&page=${randomStart}` 
}

export default apiUrls;