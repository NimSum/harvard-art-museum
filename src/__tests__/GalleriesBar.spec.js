import { mount } from '@vue/test-utils';
import GalleriesBar from '../components/GalleriesBar.vue';
import { gallery1220 } from '../utils/mockData';

describe('GalleriesBar', () => {
  const $route = {
    params: {
      floor_number: 1
    }
  }
  let wrapper;

  beforeEach(() => {
    wrapper = mount(GalleriesBar, {
      mocks: {
        $route
      },
      propsData: {
        galleries: gallery1220.records
      }
    });
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  })

  test('sets class to active on click', () => {
    wrapper.find('li').trigger('click');
    expect(wrapper.element).toMatchSnapshot();
  })
});