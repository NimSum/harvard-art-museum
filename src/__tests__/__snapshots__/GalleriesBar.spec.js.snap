// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleriesBar renders correctly 1`] = `
<aside>
  <nav>
    <h2>
      Galleries : Floor 1
    </h2>
     
    <ul>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
      <li
        class=""
      >
        
        
      
      </li>
    </ul>
  </nav>
</aside>
`;

exports[`GalleriesBar sets class to active on click 1`] = `
<aside>
  <nav>
    <h2>
      Galleries : Floor 1
    </h2>
     
    <ul>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
      <li
        class="hasVisited active"
      >
        
        
      
      </li>
    </ul>
  </nav>
</aside>
`;
